pymc_marketing/clv/models/modified_beta_geo.py

```diff
@@ -0,0 +1,316 @@
+       International Journal of Research in Marketing, 24 (3), 201-209.
+    .. [6] Wagner, U. and Hoppe D. (2008), "Erratum on the MBG/NBD Model,"
+       International Journal of Research in Marketing, 25 (3), 225-226.
+    """
+
+    _model_type = "MBG/NBD"
+
+    def build_model(self) -> None:  # type: ignore[override]
+        """Build the model."""
+        coords = {"customer_id": self.data["customer_id"]}
+        with pm.Model(coords=coords) as self.model:
+            # purchase rate priors
+            alpha = self.model_config["alpha_prior"].create_variable("alpha")
+            r = self.model_config["r_prior"].create_variable("r")
+
+            # dropout priors
+            if "a_prior" in self.model_config and "b_prior" in self.model_config:
+                a = self.model_config["a_prior"].create_variable("a")
+                b = self.model_config["b_prior"].create_variable("b")
+            else:
+                # hierarchical pooling of dropout rate priors
+                phi_dropout = self.model_config["phi_dropout_prior"].create_variable(
+                    "phi_dropout"
+                )
+                kappa_dropout = self.model_config[
+                    "kappa_dropout_prior"
+                ].create_variable("kappa_dropout")
+
+                a = pm.Deterministic("a", phi_dropout * kappa_dropout)
+                b = pm.Deterministic("b", (1.0 - phi_dropout) * kappa_dropout)
+
+            def logp(t_x, x, a, b, r, alpha, T):
+                """
+                Compute the log-likelihood of the BG/NBD model.
+
+                The log-likelihood expression here aligns with expression (4) from [3]
+                due to the possible numerical instability of expression (3).
+                """
+                a1 = pt.gammaln(r + x) - pt.gammaln(r) + r * pt.log(alpha)
+                a2 = (
+                    pt.gammaln(a + b)
+                    + pt.gammaln(b + x + 1)
+                    - pt.gammaln(b)
+                    - pt.gammaln(a + b + x + 1)
+                )
+                a3 = -(r + x) * pt.log(alpha + T)
+                a4 = (
+                    pt.log(a)
+                    - pt.log(b + x)
+                    + (r + x) * (pt.log(alpha + T) - pt.log(alpha + t_x))
+                )
+
+                logp = a1 + a2 + a3 + pt.logaddexp(a4, 0)
+
+                return check_parameters(
+                    logp,
+                    a > 0,
+                    b > 0,
+                    alpha > 0,
+                    r > 0,
+                    msg="a, b, alpha, r > 0",
+                )
+
+            pm.Potential(
+                "likelihood",
+                logp(
+                    x=self.data["frequency"],
+                    t_x=self.data["recency"],
+                    a=a,
+                    b=b,
+                    alpha=alpha,
+                    r=r,
+                    T=self.data["T"],
+                ),
+            )
+
+    def expected_num_purchases(
```

Comment:
Do we need this method just to deprecate?

Reply:
No, this should be deleted.

Reply:
If this is to be deleted, then we should as well delete the associated `BetaGeoModel.expected_num_purchases`. 

Notice that we are inheriting, and a user may by mistake call the method.

Reply:
Best to do this in a separate PR prior to merging this one, as I'm not sure if it should be added to the release notes.